src/other/utils.jl

```diff
@@ -83,3 +83,37 @@ else
 end
 
 funname(c::ComposedFunction) = Symbol(funname(c.outer), :_, funname(c.inner))
+
+"""
+    tforeach(f, x::AbstractArray; basesize::Integer)
+
+Apply function `f` to each entry in `x` in parallel, spawning
+one separate task for each block of `basesize` entries.
+
+A number of task higher than `Threads.nthreads()` may be spawned,
+since that can allow for a more efficient load balancing in case
+some threads are busy (nested parallelism).
+"""
+function tforeach(f, x::AbstractArray; basesize::Integer)
```

Comment:
This docstring makes Documentation build fail as it's not included in the manual. I guess I can just turn it into a comment if we don't have a better solution.

Reply:
We have a solution for this.
Add this function to docs/src/lib/internals.md